generator/src/main/java/com/scottlogic/deg/generator/restrictions/ParsedGranularity.java

```diff
@@ -19,6 +19,10 @@ public static ParsedGranularity parse(Object granularityExpression) {
         if (granularityExpression instanceof Number) {
             BigDecimal asNumber = NumberUtils.coerceToBigDecimal(granularityExpression);
 
+            if (asNumber == null){
+                throw new IllegalArgumentException("Numeric granularity input type is not supported");
```

Comment:
So my assumption is that if NumberUtils can't "coerce" this it returns null? Why is this exception not thrown by the coerce method?

Reply:
Because its the responsibility of the calling code to made the decisions as to whether a `null` response is exceptional